src/invariant/EventsAreConsistentWithEntryDiffs.cpp

```diff
@@ -356,9 +359,37 @@ calculateDeltaBalance(AggregatedEvents& agg,
         };
         auto entryDiff =
             rust_bridge::i128_sub(getAmount(current), getAmount(previous));
-        return entryDiff == eventDiff
-                   ? ""
-                   : "ContractData diff does not match events";
+
+        if (entryDiff == eventDiff)
+        {
+            return "";
+        }
+
+        // Before reporting an error, check if there's a reconciliation event
+        // from p23_hot_archive_bug that accounts for the difference
+        // This isn't a perfect check, but it gives us reasonable confidence.
+        if (maybeAddress &&
+            protocolVersionEquals(protocolVersion, ProtocolVersion::V_23))
+        {
+            auto const& reconciler =
+                app.getProtocol23CorruptionEventReconciler();
+            if (reconciler)
+            {
+                // The difference between eventDiff and entryDiff should be
+                // accounted for by a reconciliation event
+                auto diff = rust_bridge::i128_sub(eventDiff, entryDiff);
+
+                if (reconciler->hasReconciliationAmount(asset, *maybeAddress,
```

Comment:
This seems a bit sketchy at a first glance, though I assume it works fine. Maybe it's worth adding a comment stating that the affected balances have not been interacted with much and that the amounts minted/burned are unique to the corruption events.

Reply:
Updated the comment. This isn't meant to be an exact check because I didn't think one was necessary.